local-server/local-server-services/src/main/java/com/esri/samples/localserver/LocalServerServicesController.java

```diff
@@ -0,0 +1,287 @@
+   * <p>
+   * Only creates a Local Service that is not currently  running on the Local Server.
+   * 
+   * @param event the event that fired this action
+   * @throws URISyntaxException if URL passed to Local Service is incorrect
+   */
+  @FXML
+  protected void handleStartService(ActionEvent event) throws URISyntaxException {
+
+    if (server.getStatus() == LocalServerStatus.STARTED) {
+      // check that the service selected is not already running
+      boolean serviceExists = false;
+      String serviceSelected = serviceOptions.getSelectionModel().getSelectedItem();
+      for (String runningService : runningServices.getItems()) {
+        if (runningService.startsWith(serviceSelected)) {
+          serviceExists = true;
+        }
+      }
+
+      // start service if not currently running on the Local Server
+      if (!serviceExists) {
+        String serviceUrl = "";
+        LocalService localService = null;
+        switch (serviceSelected) {
+          case "Map Service":
+            serviceUrl = Paths.get(getClass().getResource("/PointsofInterest.mpk").toURI()).toString();
+            localService = new LocalMapService(serviceUrl);
+            break;
+          case "Feature Service":
+            serviceUrl = Paths.get(getClass().getResource("/PointsofInterest.mpk").toURI()).toString();
+            localService = new LocalFeatureService(serviceUrl);
+            break;
+          case "Geoprocessing Service":
+            serviceUrl = Paths.get(getClass().getResource("/MessageInABottle.gpk").toURI()).toString();
+            localService = new LocalGeoprocessingService(serviceUrl);
+        }
+
+        startLocalService(localService, serviceSelected);
+
+      } else {
+        // if service is currenlt running on server warn user
+        Platform.runLater(() -> {
+          Alert dialog = new Alert(AlertType.INFORMATION);
+          dialog.setHeaderText("Service Running");
+          dialog.setContentText(serviceSelected + " has already been started.");
+          dialog.showAndWait();
+        });
+      }
+    }
+  }
+
+  /**
+   * Starts a Local Service.
+   * <p>
+   * The status of the Local Service is displayed in the center text area of the application.
+   * The URL of the Local Service is added to the list of running services.
+   * 
+   * @param localService Local Service to start.
+   * @param serviceName Name of Local Service. 
+   */
+  private void startLocalService(LocalService localService, String serviceName) {
+    localService.addStatusChangedListener(status -> {
+      centerText.appendText(serviceName + " Status: " + status.getNewStatus().toString() + "\n");
+      if (status.getNewStatus() == LocalServerStatus.STARTED) {
+        Platform.runLater(() -> runningServices.getItems().add(serviceName + " URL ->  " + localService.getUrl()));
+
+      } else if (isServerStopping && status.getNewStatus() == LocalServerStatus.STOPPED) {

```

Comment:
@colinanderson 
Here is the code that I added to stop the services and server how I wanted.